internal/schemautil/custom_diff.go

```diff
@@ -320,3 +326,89 @@ func CustomizeDiffAdditionalDiskSpace(ctx context.Context, diff *schema.Resource
 	// which otherwise will be suppressed by TF
 	return diff.SetNew(k, "0B")
 }
+
+// CustomizeDiffCheckPlan validates that the service plan exists and suggests similar plans if not found
+func CustomizeDiffCheckPlan(ctx context.Context, d *schema.ResourceDiff, _ interface{}) error {
+	client, err := common.GenClient()
+	if err != nil {
+		return err
+	}
+
+	var (
+		project     = d.Get("project").(string)
+		serviceType = d.Get("service_type").(string)
+		plan        = d.Get("plan").(string)
+
+		maxSuggestions = 3
+	)
+
+	if project == "" || serviceType == "" || plan == "" {
+		return nil
+	}
+
+	plans, err := client.ProjectServicePlanList(ctx, project, serviceType)
+	if err != nil {
+		return fmt.Errorf("unable to fetch service plans: %w", err)
+	}
```

Comment:
I think, we can use there `schemautil.DoOnce` here (key project+serviceType) to lower the pressure on the API and speed up the check a little (considering the authorization, authentication and the list fetch).